Add Archetype Version to archetype.properties file

pat-lego opened the pull request:
As a developer, it is important to know which archetype version was used since it impacts the ability to deploy the project to an AEM instance.

## Description

Add archetypeVersion to the archetype.properties file

## Related Issue

No related issue just something I wanted to add to the project since it was bothering me that it was not already part of the archetype.properties file.

## Motivation and Context

I am always documenting the archetype version used in the README but I would much prefer having it in the archetype.properties file since it already contained much of the archetype options that were entered within the project.

## How Has This Been Tested?

 - Added the variable to the archetype-metadata.xml
 - Ran a `mvn clean install`
 - All tests passed
 - Created a new project from the SNAPSHOT build that was previously built
 - Validated that the archetype.properties file was properly created

## Screenshots (if appropriate):

![Screen Shot 2021-04-30 at 2 11 05 PM](https://user-images.githubusercontent.com/36930772/116736527-fe94a680-a9bd-11eb-9482-0debe55d472b.png)
![Screen Shot 2021-04-30 at 2 11 14 PM](https://user-images.githubusercontent.com/36930772/116736530-ff2d3d00-a9bd-11eb-99ba-bc88ff2aa063.png)
![Screen Shot 2021-04-30 at 2 11 20 PM](https://user-images.githubusercontent.com/36930772/116736531-ff2d3d00-a9bd-11eb-829b-1822e2ed5df3.png)
![Screen Shot 2021-04-30 at 2 11 28 PM](https://user-images.githubusercontent.com/36930772/116736532-ff2d3d00-a9bd-11eb-9ffa-b11d5d69feb2.png)

## Types of changes

<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->

- [x] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

pat-lego committed:
Adding archetypeVersion in the properties

pat-lego committed:
Adding archetype properties

pat-lego committed:
Changing comment value

pat-lego commented:
@sharanyavinod would you be able to review this. I am much looking forward to this functionality.

pat-lego committed:
Merge branch 'develop' into feature/resolve-archetype-version-pr

vladbailescu commented on src/test/resources/projects/basic/archetype.properties:
```diff
@@ -20,4 +20,5 @@ sdkFormsVersion=latest
 datalayer=y
 amp=y
 enableDynamicMedia=y
+archetypeVersion=28-SNAPSHOT
```
Will this need to be updated manually after each release? Can we make it optional? It will always be present from the invocation anyway

pat-lego committed:
Adding default value

This will prevent us from having to change the test every time we can assert against the default value to make sure it comes through
